src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/Composition.java

```diff
@@ -149,6 +150,37 @@ public void setChildren(List<Composable> children, ErrorStatus errorStatus) {
 
     public native HashMap<Composable, TimeRange> getRangeOfAllChildren(ErrorStatus errorStatus);
 
+    public <T extends Composable> Stream<T> eachChild(
+            TimeRange searchRange, Class<T> descendedFrom, boolean shallowSearch, ErrorStatus errorStatus) {
+        List<Composable> children;
+        if (searchRange != null) {
+            children = this.getChildren();
+        } else {
+            children = this.getChildren();
+        }
+
```

Comment:
possible simplification of 162-177

```
return Stream.concat(
children.stream(), 
if (element.getClass().isAssignableFrom(descendedFrom))
   Stream.of(descendedFrom.cast(element))
if (!shallowSearch && element instanceof Composition)
   ((Composition) element).eachChild(
                                        searchRange == null ? null : this.getTransformedTimeRange(searchRange, ((Composition) element), errorStatus),
                                        descendedFrom,
                                        shallowSearch,
                                        errorStatus);
else 
   Stream.empty() 
);
```

Reply:
@margant I dont understand how this simplification works. There's no map function.